### PR TITLE
Usability improvements for follow_plugin (wildlife)

### PR DESCRIPTION
**Pull request by j-herman**
Substantive changes are all in follow_plugin.cc and follow_plugin.hh - remaining files are simply duplicates of the animal models without the plugin attached to support other efforts.
To confirm backwards compatibility, simply run any existing wildlife example with a hard-coded waypoint path and confirm the animal buoy moves as expected.
To test the new functionality of the plugin, modify the model.sdf for one of the animals, replacing the `<waypoints>` section in the follow plugin with:
```
<line>
  <direction>360</direction>
  <length>10</length>
</line>
```
Rerun the wildlife example and confirm the selected animal travels east about 10 meters and then reverses direction.  

Now replace the `<line>` section with:
```
<circle>
<radius>2</radius>
</circle>
```
Rerun the example and confirm the animal travels in a small circle, radius about 2 meters.

### CODE REVIEW

File: vrx_gazebo/models/crocodile_static/model.sdf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        <pose>0 0 0.06 0 0 0</pose>
+        <geometry>
+          <box>
+            <size>0.80 0.40 0.15</size>
+          </box>
+        </geometry>
+      </collision>
+      <visual name="crocodile_visual">
+        <geometry>
+	        <mesh><uri>model://crocodile_buoy/meshes/crocodile.dae</uri></mesh>
```
Comment (caguero) [inline vrx_gazebo/models/crocodile_static/model.sdf:30]: This is pointing to the non-static version, which is totally fine. I think you can remove all the `.dae`, `.png` and `.material` from the static versions. It should be just the `model.sdf` and the `model.config` reusing all the resources from the non-static version for example.

**Review by caguero — CHANGES_REQUESTED** [review]

File: vrx_gazebo/models/crocodile_static/model.sdf (hunk 1 above)
Comment (j-herman) [reply vrx_gazebo/models/crocodile_static/model.sdf:30]: Removed files and double-checked for correctness

**Review by caguero — COMMENTED** [review]
I commit some changes (mostly stylish) in 372a4c3.

File: vrx_gazebo/include/vrx_gazebo/follow_plugin.hh [hunk 2]
```diff
@@ -49,6 +51,11 @@ namespace gazebo
 ///              the model should navigate through. This block should contain at
 ///              least one of these blocks:
 ///                <waypoint>: This block should contain the X, Y of a waypoint.
+/// <line>: Element that indicates the model should travel in "line" mode.
+///         The block should contain the relative direction from the initial
+///         position in which the vehicle should move, specified in the world
+///         frame.
```
Comment (caguero) [inline vrx_gazebo/include/vrx_gazebo/follow_plugin.hh:57]: Could you specify the units? Could you document the meaning of `<distance>` and its units in this context?

File: vrx_gazebo/include/vrx_gazebo/follow_plugin.hh [hunk 3]
```diff
@@ -49,6 +51,11 @@ namespace gazebo
...
+/// <circle>: Element that indicates the model should travel in "circle" mode.
```
Comment (caguero) [inline vrx_gazebo/include/vrx_gazebo/follow_plugin.hh:58]: Could you document the meaning of `<distance>` and its units in this context?

**Review by caguero — CHANGES_REQUESTED** [review]
My understanding is that the `<distance>` parameter only makes sense if you're using `<circle>` or `<line>`. Also, if you're using `circle`, there are no paramers inside this element. I recommend the following change in SDF to make it more clear:

```
<circle>
  <radius>5</radius>
</circle>
```
and
```
<line>
  <angle>2</angle>
  <length>10</length>
</line>
```

**Review by caguero — COMMENTED** [review]
Could you also add to the `follow_plugin.hh` documentation two more examples that capture how to use the `<line>` and `<circle>` options?

**Comment by j-herman** [comment]
@caguero I think I've gotten all of your comments addressed.  Updated directions for checking the plugin functionality are included above.

File: vrx_gazebo/src/follow_plugin.cc [hunk 4]
```diff
@@ -64,7 +70,66 @@ void FollowPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
...
+    {
+      auto radElem = circleElem->GetElement("radius");
+      radius = radElem->Get<double>();
+    }
+    // Get the current model position in global coordinates.  Create
+    // local vectors that represent a path along a rough circle.
+    ignition::math::Vector2d position(this->modelPose.Pos().X(),
+                                      this->modelPose.Pos().Y());
+    double angle = 0;
+    ignition::math::Vector2d vec(radius / 2, 0);
```
Comment (caguero) [inline vrx_gazebo/src/follow_plugin.cc:90]: @j-herman , I know we merge this but does it make sense `radius / 2` here and in line 97?  I think it should be just radius, right?

Comment (j-herman) [reply vrx_gazebo/src/follow_plugin.cc:90]: Oh, no!  I started out with circumference measurements in the initial version and never updated the equation.  I'm inclined to fix it by changing the variable name and documentation back, so nothing will change with the existing worlds - sound ok?

Comment (caguero) [reply vrx_gazebo/src/follow_plugin.cc:90]: > Oh, no! I started out with circumference measurements in the initial version and never updated the equation. I'm inclined to fix it by changing the variable name and documentation back, so nothing will change with the existing worlds - sound ok?

So converting it to `diameter`?

Comment (j-herman) [reply vrx_gazebo/src/follow_plugin.cc:90]: Diameter.  Right.  Not circumference.

Comment (caguero) [reply vrx_gazebo/src/follow_plugin.cc:90]: Sounds good to me!

**Pull request merged**